src/backend/joanie/tests/core/api/admin/course_product_relations/test_generate_certificates.py

```diff
@@ -0,0 +1,770 @@
+            course=course,
+            enrollment_end=timezone.now() + datetime.timedelta(hours=1),
+            enrollment_start=timezone.now() - datetime.timedelta(hours=1),
+            is_gradable=True,
+            start=timezone.now() - datetime.timedelta(hours=1),
+        )
+        factories.ProductTargetCourseRelationFactory(
+            product=product,
+            course=course,
+            is_graded=True,
+        )
+
+        cpr = CourseProductRelation.objects.get(product=product, course=course)
+
+        # Create some certificates that are already generated
+        orders_in_past = factories.OrderFactory.create_batch(
+            5,
+            product=cpr.product,
+            course=cpr.course,
+        )
+        for order in orders_in_past:
+            order.submit()
+            factories.OrderCertificateFactory(order=order)
+
+        self.assertEqual(Certificate.objects.count(), 5)
+
+        # Create orders where we will generate certificates
+        orders = factories.OrderFactory.create_batch(
+            10,
+            product=cpr.product,
+            course=cpr.course,
+        )
+        for order in orders:
+            order.submit()
+
+        mock_generate_certificates_task.delay.return_value = ""
+
+        with mock.patch.object(DummyLMSBackend, "get_grades") as mock_get_grades:
+            mock_get_grades.return_value = {"passed": True}
+
+            response = self.client.post(
+                f"/api/v1.0/admin/course-product-relations/{cpr.id}/generate_certificates/",
+                content_type="application/json",
+            )
+            self.assertTrue(mock_generate_certificates_task.delay.called)
+
+        expected_cache_data_response = {
+            "in_progress": True,
+            "course_product_relation_id": str(cpr.id),
+            "count_certificate_to_generate": len(orders),
+            "count_exist_before_generation": 5,
+        }
+
+        self.assertEqual(response.status_code, HTTPStatus.CREATED)
+        self.assertDictEqual(response.json(), expected_cache_data_response)
+
+        # When the endpoint is requested again with the same course product relation id
+        second_response = self.client.post(
+            f"/api/v1.0/admin/course-product-relations/{cpr.id}/generate_certificates/",
+            content_type="application/json",
+        )
+
+        self.assertEqual(second_response.status_code, HTTPStatus.ACCEPTED)
+        self.assertDictEqual(second_response.json(), expected_cache_data_response)
+
+        cache_key = f"celery_certificate_generation_{cpr.id}"
+        cache_data = cache.get(cache_key)
+        self.assertDictEqual(cache_data, expected_cache_data_response)
+
+    @override_settings(CELERY_TASK_ALWAYS_EAGER=False)
+    @mock.patch("joanie.core.api.admin.generate_certificates_task")
+    def test_admin_api_course_product_relation_generate_certificates_authenticated_task_launched(
+        self, mock_generate_certificates_task
+    ):
```

Comment:
This test and the one before can be merged IMO. You can check the cache in the same test.